### ISSUE TITLE
AzKeyVault v 4.02-preview REGRESSION - Register-SecretVault: Could not find a SecretManagement extension implementing script module.

### ISSUE THREAD

**Issue by TheBigBear** [PowerShell/SecretManagement#104]
I am trying to add and use an Azure Keyvault. 

If I use: 

```
Register-SecretVault -Name AzKeyVault -ModuleName Az.KeyVault -VaultParameters @{ AZKVaultName = 'My-AzKeyVault'; SubscriptionId = '8a2e...-...-...-...-43fbcfe7...' } # subscriptionId redacted here
```

and I get the following error in red

```
Register-SecretVault: Could not find a SecretManagement extension implementing script module.
```

What did I do wrong? Pretty sure this used to work in an earlier preview of the module? 

PS: I have the following versions of modules installed. 

```
Version                    Name                              
-------                    ----                                
4.0.2-preview       Az.KeyVault                         
0.9.0                        Microsoft.PowerShell.SecretStore
0.9.0                        Microsoft.PowerShell.SecretManagement

```

**Comment by TheBigBear** [comment]
But if I remove Az.Keyvault version 4.0.2-preview and install the last stable released version 3.4.0 instead it all still works. So somehow the two modules have a regression?

```
Set-Secret -Vault AzKeyVault -Name test-secret-1 -Secret 'test-secret-1'
```

```
Get-Secret -Vault AzKeyVault -name test-secret-1
System.Security.SecureString
```

**Comment by PaulHigin** [comment]
The Az.Keyvault is not a SecretManagement extension module.  It appears to just be a standard PowerShell module for using AzKeyVault.  I believe the team is working on a SecretManagement extension module, but don't know if there is a preview available.

To find all SecretManagement extension modules, you can search PSGallery using the `SecretManagement` tag:

```powershell
PS C:\Scripts> Find-Module -Tag 'SecretManagement' -Repository PSGallery

Version              Name                                Repository           Description
-------              ----                                ----------           -----------
0.9.1                Microsoft.PowerShell.SecretStore    PSGallery            This PowerShell module is an extension v…
0.9.1.1              SecretManagement.KeePass            PSGallery            A cross-platform Keepass Secret Manageme…
0.2.1                SecretManagement.LastPass           PSGallery            SecretManagement extension for LastPass!
0.1.1                SecretManagement.BitWarden          PSGallery            SecretManagement extension for BitWarden!
0.1.3                SecretManagement.KeyChain           PSGallery            SecretManagement extension vault for mac…
0.0.4.5              SecretManagement.1Password          PSGallery            SecretManagement extension for 1Password
0.2.351              SecretManagement.PleasantPasswordS… PSGallery            A cross-platform Pleasent Password Serve…
1.0.2                SecretManagement.Keybase            PSGallery            Keybase Secret Management Extension
0.0.9.1              SecretManagement.Chromium           PSGallery            A cross-platform Chromium (Edge/Chrome) …
```
